### PR TITLE
Feature/241 get rid of analysis config difference between with and without network

### PR DESCRIPTION
**Pull request by ArdtK**
@Carsopre main changes (single commit per change):
* Replace AnalysisConfigData* with a single class AnalysisConfigData
* Replace AnalysisConfigReader* with a single class AnalysisConfigDataReader (took AnalysisConfigReaderBase as basis)
  * reading the file network config from network.ini is moved to the wrapper, which felt like a more logical place
* Replace AnalysisConfigValidator* with a single class AnalysisConfigValidator (took AnalysisConfigValidatorWithoutNetwork as basis)
  * as network.ini is not read anymore in the AnalysisConfigDataReader checks on existence of network.ini is removed from the validator
* Replace AnalysisConfigWrapper* with a single class AnalysisConfigWrapper (took AnalysisConfigWrapperWithNetwork as basis)
  * a new method from_data_without_network is added
  * choose between from_data_with/without_network method in config_factory
* Renamed main to __main__ to allow running ra2ce as a package
 Tests are adjusted accordingly.

### CODE REVIEW

File: ra2ce/__main__.py
Comment (Carsopre) [inline ra2ce/__main__.py:1]: I don't think this was in the scope of this issue. But fine.

File: ra2ce/analyses/analysis_config_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    @staticmethod
+    def get_data_output(ini_file: Path) -> Optional[Path]:
+        _root_path = AnalysisConfigWrapper.get_network_root_dir(ini_file)
+        _project_name = ini_file.parent.name
+        return _root_path.joinpath(_project_name, "output")
+
+    @property
+    def root_dir(self) -> Path:
+        return self.get_network_root_dir(self.ini_file)
```
Comment (Carsopre) [inline ra2ce/analyses/analysis_config_wrapper.py:61]: Would be ideal if we could move this logic into the `AnalysisConfigData` and leave the wrapper as just a 'controller' of the configuration `initialize_output_dirs` and so  on.

File: ra2ce/analyses/analysis_config_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        _new_analysis.graphs = network_config.graphs
+
+        return _new_analysis
+
+    @classmethod
+    def from_data_without_network(
+        cls,
+        ini_file: Path,
+        config_data: AnalysisConfigData,
+    ) -> AnalysisConfigWrapper:
```
Comment (Carsopre) [inline ra2ce/analyses/analysis_config_wrapper.py:131]: Would it make sense to get some of this logic into the factory so that here we only have `from_data_with_network`? Let's aim to move all this logic to the factory so the responsibility to initialize the wrappers lays there.

File: ra2ce/graph/network_config_wrapper.py [hunk 3]
```diff
@@ -151,6 +156,10 @@ def read_graphs_from_config(static_output_dir: Path) -> dict:
 
         return _graphs
 
+    def configure(self) -> None:
```
Comment (Carsopre) [inline ra2ce/graph/network_config_wrapper.py:159]: Well seen.

File: tests/test_data/acceptance_test_data/output/analyses.ini
Comment (Carsopre) [inline tests/test_data/acceptance_test_data/output/analyses.ini:1]: Why was this file added?

File: ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py [hunk 4]
```diff
@@ -65,11 +73,11 @@ def read(self, ini_file: Path) -> AnalysisConfigData:
             output_path=_parent_dir.joinpath("output"),
             **self._get_sections(),
         )
-        _config_data.root_path = AnalysisConfigWrapperBase.get_network_root_dir(
-            ini_file
-        )
+        _config_data.root_path = AnalysisConfigWrapper.get_network_root_dir(ini_file)
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py:76]: Update, although we can't reach exactly what I would like as we need an `output_dir` to dump the log we can at least do the following:
- `get_network_root_dir` is not really being used (only here in a direct way), you can remove the function and set the value directly in the `AnalysisConfigData` constructor: `AnalysisConfigData(root_path=ini_file.parent.parent, ...)`
- `get_data_output` actually translates to `return ini_file.parent.joinpath("output")`, update accordingly.
- `root_dir` translates to `self.ini_file.parent.parent` update accordingly.

**Review by Carsopre — CHANGES_REQUESTED** [review]
Almost there, just a few remarks.

File: ra2ce/__main__.py
Comment (ArdtK) [reply ra2ce/__main__.py:1]: Agree. I needed this to be able to run ra2ce as package, as done in the old-fashioned acceptance test of Koswat.

File: ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py:76]: Clear. 1 comment:
`return ini_file.parent.joinpath("output")`
should be
`return ini_file.parent.joinpath(_project_name, "output")`

[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py:76]: I adjusted the same functionality in Graph as well

File: tests/test_data/acceptance_test_data/output/analyses.ini
Comment (ArdtK) [reply tests/test_data/acceptance_test_data/output/analyses.ini:1]: Hmm, by mistake, I guess. I'll remove it again.

File: ra2ce/analyses/analysis_config_wrapper.py (hunk 1 above)
Comment (ArdtK) [reply ra2ce/analyses/analysis_config_wrapper.py]: Done

File: ra2ce/analyses/analysis_config_wrapper.py (hunk 2 above)
Comment (ArdtK) [reply ra2ce/analyses/analysis_config_wrapper.py]: Done

**Review by ArdtK — COMMENTED** [review]
@Carsopre I processed your review comments. Thoroughly tried to cleanup logic regarding the root_dir.

File: ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py [hunk 5]
```diff
@@ -208,7 +202,7 @@ def _get_analysis_section_direct(self, section_name: str) -> AnalysisSectionDire
             "evaluation_period",
             fallback=_section.evaluation_period,
         )
-        _section.return_pinterest_rateeriod = self._parser.getfloat(
+        _section.interest_rate = self._parser.getfloat(
```
Comment (ArdtK) [inline ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py:205]: Copy/paste error corrected

Comment (Carsopre) [reply ra2ce/analyses/analysis_config_data/analysis_config_data_reader.py:205]: well seen :)

File: ra2ce/configuration/config_factory.py [hunk 6]
```diff
@@ -66,7 +64,7 @@ def get_config_wrapper(
         return _input_config
 
     @staticmethod
-    def get_network_config_data(network_ini: Path) -> Optional[NetworkConfigData]:
+    def get_network_config_data(network_ini: Path) -> Optional[NetworkConfigWrapper]:
```
Comment (Carsopre) [inline ra2ce/configuration/config_factory.py:67]: This returns a `NetworkConfigData`, not a wrapper.

**Review by Carsopre — APPROVED** [review]

File: ra2ce/configuration/config_factory.py (hunk 6 above)
Comment (ArdtK) [reply ra2ce/configuration/config_factory.py:67]: Not sure if I agree
